src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/FileIO/SpecialDirectories.vb

```diff
@@ -0,0 +1,286 @@
+        ''' </summary>
+        ''' <value>A String containing the path to the current user's Desktop directory.</value>
+        ''' <remarks>This directory is C:\Document and Settings\[UserName]\Desktop.</remarks>
+        Public Shared ReadOnly Property Desktop() As String
+            Get
+                Return GetDirectoryPath(Environment.GetFolderPath(SpecialFolder.Desktop), SR.IO_SpecialDirectory_Desktop)
+            End Get
+        End Property
+
+        ''' <summary>
+        ''' Returns the directory used to store program shortcuts from Start Menu for current user.
+        ''' </summary>
+        ''' <value>A String containing the path to the Start Menu \ Programs directory.</value>
+        ''' <remarks>This directory is C:\Document and Settings\[UserName]\Start Menu\Programs.</remarks>
+        Public Shared ReadOnly Property Programs() As String
+            Get
+                Return GetDirectoryPath(Environment.GetFolderPath(SpecialFolder.Programs), SR.IO_SpecialDirectory_Programs)
+            End Get
+        End Property
+
+        ''' <summary>
+        ''' Return the program files directory.
+        ''' </summary>
+        ''' <value>A String containing the path to the default program directories.</value>
+        ''' <remarks>This directory is C:\Program Files.</remarks>
+        Public Shared ReadOnly Property ProgramFiles() As String
+            Get
+                Return GetDirectoryPath(Environment.GetFolderPath(SpecialFolder.ProgramFiles), SR.IO_SpecialDirectory_ProgramFiles)
+            End Get
+        End Property
+
+        ''' <summary>
+        ''' Return the directory that contain temporary files for the current user.
+        ''' </summary>
+        ''' <value>A String containing the path to the temporary directory for the current user.</value>
+        ''' <remarks>
+        ''' According to Win32 API document, GetTempPath should always return a value even if TEMP and TMP = "".
+        ''' Also, this is not updated if TEMP or TMP is changed in Windows. The reason is
+        '''     each process has its own copy of the environment variables and this copy is not updated.
+        ''' </remarks>
+        Public Shared ReadOnly Property Temp() As String
+            Get
+                Return GetDirectoryPath(IO.Path.GetTempPath(), SR.IO_SpecialDirectory_Temp)
+            End Get
+        End Property
+
+        ''' <summary>
+        ''' Returns the directory that serves as a common repository for data files
+        ''' from your application used only by the current user.
+        ''' </summary>
+        ''' <value>A String containing the path to the directory your application can use to store data for the current user.</value>
+        ''' <remarks>
+        ''' We chose to use UserAppDataPath instead of LocalUserAppDataPath since this directory
+        ''' will work with Roaming User as well.
+        ''' </remarks>
+        Public Shared ReadOnly Property CurrentUserApplicationData() As String
```

Comment:
>CurrentUserApplicationData [](start = 40, length = 26)

There are many changes to `CurrentUserApplicationData` and `AllUsersApplicationData` from the reference source. Is that because `System.Windows.Forms` is not available on .NET Core yet? If so, consider excluding these methods from this PR so that this PR contains a straight port of the reference source with as few changes as possible.

Reply:
Yes System.Windows.Forms is not available (and don't know if they will be or when) and why I changed it to WIP, I have it working exactly on Windows and the logic for Unix I just need to test and refine. If there is a hurry for the rest, I can Throw PlatformNotSupported and do another PR later but I think these are needed functions and they work every different then Environment.SpecialFolders.